tox.ini

```diff
@@ -1,31 +1,55 @@
 [tox]
 envlist = 
-    py{38}
+    tests
+    linting
+    coverage
     docs
-    flake8
 skip_missing_interpreters = True
 isolated_build = True
 
 [testenv]
-deps =
-    -r requirements/tests.txt
-
-commands = pytest --basetemp={envtmpdir} {posargs}
-
+commands = python -m pip install --upgrade pip
 
 [testenv:dev]
 description = development environment
-basepython = python3.8
 usedevelop = True
+deps =
+    black
+    pip-tools
+    flake8
+    flakehell
+    pytest
+    pytest-datafiles
 commands = 
     python -m pip install --upgrade pip
-    pip install -q pip-tools
     pip-compile requirements/base.in 
     pip-compile requirements/dev.in 
     pip-compile requirements/docs.in 
     pip-compile requirements/tests.in
     pip install -q -e . -r requirements/dev.txt
 
+[testenv:tests]
+deps =
+    -rrequirements/base.txt
+    pytest
+    pytest-datafiles
+commands = python -m pytest --basetemp={envtmpdir} {posargs}
+
+[testenv:linting]
+skip_install = true
```

Comment:
Was tut das `skip_install`?

Reply:
From the [documentation](https://tox.readthedocs.io/en/latest/config.html#conf-skip_install):
>**skip_install**=false(true|false)
>Do not install the current package. This can be used when you need the virtualenv management but do not want to install the current package into that environment.

Since we are creating our own python package here, you could install it in the appropriate virtual environment.
However, this is not necessary in the linting env. Therefore we skip this.